src/app/child-dev-project/attendance/add-day-attendance/roll-call/roll-call.component.ts

```diff
@@ -149,24 +67,17 @@ export class RollCallComponent implements OnInit {
     } else {
       this.currentIndex++;
     }
-
-    this.displayRecordForDebugging(this.rollCallList[this.currentIndex]);
   }
 
   endRollCall() {
-    this.complete.emit();
+    this.complete.emit(this.eventEntity);
   }
 
   isFinished(): boolean {
-    return this.currentIndex >= this.rollCallList.length;
+    return this.currentIndex >= this.eventEntity.children.length;
   }
 
-  displayRecordForDebugging(rollCallRecord: RollCallRecord) {
-    if (!rollCallRecord) {
-      return;
-    }
-
-    console.log(rollCallRecord);
-    this.debugContent = JSON.stringify(rollCallRecord.attendanceDay, null, 2);
+  getChildById(childId: string) {
```

Comment:
In the construction of the `entries` object, the child object could already be linked (instead of the `chidlId`). This would make this function unnecessary.